apps/meteor/app/reactions/server/setReaction.ts

```diff
@@ -100,24 +78,32 @@ async function setReaction(room: IRoom, user: IUser, message: IMessage, reaction
 		await Messages.setReactions(message._id, message.reactions);
 		if (isTheLastMessage(room, message)) {
 			await Rooms.setReactionsInLastMessage(room._id, message.reactions);
-			void notifyOnRoomChangedById(room._id);
 		}
-		await callbacks.run('setReaction', message._id, reaction);
-		await callbacks.run('afterSetReaction', message, { user, reaction, shouldReact });
+		void callbacks.run('afterSetReaction', message, { user, reaction, shouldReact: true });
 
 		isReacted = true;
 	}
 
-	await Apps.self?.triggerEvent(AppEvents.IPostMessageReacted, message, user, reaction, isReacted);
+	void Apps.self?.triggerEvent(AppEvents.IPostMessageReacted, message, user, reaction, isReacted);
```

Comment:
Any reason for that being `await`ed here before? 🤔

Reply:
No reason, none of the actions in here can prevent a reaction from being set (or even change something from the reaction).
Callbacks as well were only used by federation & slackbridge so probably nothing we want to be awaiting.